### ISSUE TITLE
Memory usage goes through the roof when I use pyls and don't restart emacs for a few days. 

### ISSUE THREAD

**Issue by thomasf** [emacs-lsp/lsp-mode#772]
**Describe the bug**

I have had emacs RSS size grow to well over 16GB (and continually increasing) after running a few days with a lot of pyls processes which uses about 1-2gb each.. Not much to do about the pyls processes I guess but I can't find whats taking up all the memory in Emacs.. I've tried looking through all allocated elisp objects, open fd's etc but I have not found anything suspiciously large that way.  Without pyls and lsp-mode my emacs'es usually tops out at between 1-2GB ram. 

I'm not sure how to debug this. I have also started using lsp-mode for Go code but I have just edited smaller Go projects and not for an extended time so I cannot say if happens there as well since I usually close those emacs instances. 

I normally use one emacs running emacs process per "project" where a project is any combination of code reposiories related to one customer so the 16GB emacs process was just one of several emacses I had runnin at the time.

I just upgraded to emacs 26.2, the behavior was observed under emacs 26.1 a couple of weeks ago so I don't know if there are any relevant emacs or lsp-mode bugs behind this which might have been solved since then.

**Comment by yyoncho** [comment]
Can you try this :https://lists.gnu.org/archive/html/help-gnu-emacs/2004-09/msg00342.html

**Comment by thomasf** [comment]
Yeah, I have that package from melpa and I check it and it just listed a few hundreds of mb of ram usage there IIRC, at least far far below the 16GB ram emacs was using.. I also tried to investigate all symbols myself without finding anything relevant on the visible side. I have comparably to the defaults very lax garbage collection settings but I could see that the GC was running now and then so it should not be the reason. I also tried to temporarily default the values without any change in behaviour.

```el
(setq-default ;; alloc.c
 gc-cons-threshold (* 20 1204 1204)
 gc-cons-percentage 0.5)
```
 
I'll try to collect some more data in this issue if I am working on a single project for an extended time and this happens again. It's hard to create a reproducable test case from this and it is of course possible that it's one of the other 500+ melpa packages in my 10kloc+ `init.el` that  has something to do with this but I can see a correlation between rising memory usage and `lsp-mode` being active, for laguages other than python and Go I still use auto-complete-mode + various stuff and it doesn't seem be a problem when working on non go/python projects.
